namespace/namespace.go

```diff
@@ -147,23 +147,34 @@ func (n Namespace) Repeat(times int) []Namespace {
 }
 
 func (n Namespace) Equals(n2 Namespace) bool {
-	return bytes.Equal(n.Bytes(), n2.Bytes())
+	return n.Version == n2.Version && bytes.Equal(n.ID, n2.ID)
 }
 
 func (n Namespace) IsLessThan(n2 Namespace) bool {
-	return bytes.Compare(n.Bytes(), n2.Bytes()) == -1
+	return n.Compare(n2) == -1
 }
 
 func (n Namespace) IsLessOrEqualThan(n2 Namespace) bool {
-	return bytes.Compare(n.Bytes(), n2.Bytes()) < 1
+	return n.Compare(n2) < 1
 }
 
 func (n Namespace) IsGreaterThan(n2 Namespace) bool {
-	return bytes.Compare(n.Bytes(), n2.Bytes()) == 1
+	return n.Compare(n2) == 1
 }
 
 func (n Namespace) IsGreaterOrEqualThan(n2 Namespace) bool {
-	return bytes.Compare(n.Bytes(), n2.Bytes()) > -1
+	return n.Compare(n2) > -1
+}
+
+func (n Namespace) Compare(n2 Namespace) int {
+	switch {
+	case n.Version == n2.Version:
+		return bytes.Compare(n.ID, n2.ID)
+	case n.Version < n2.Version:
+		return -1
```

Comment:
Later today I will check that -1 and 1 are not mixed here 🙈

Reply:
One way to test is to add a unit test for `Compare` and ensure that the output from before this change matches the output from after this PR because AFAIU this is strictly a performance optimization but behavior should remain the same.

Reply:
Done, `Test_compareMethods` covers all code branches now.